### ISSUE TITLE
Goal: 100% of JS semantics can be accurately represented in the type system?

### ISSUE THREAD

**Issue by ljharb** [tc39/proposal-type-annotations#37]
In other words, to name a few examples from our favorite typed alternative language:
 - null prototypes
 - indexable symbol keys
 - `[].concat()` and similar idioms
 - obscure-yet-idiomatic call/bind/apply hijinx
 - approximating internal slot checks for builtins (not instanceof semantics, which don't work cross-realm), eg, "a Date object from another realm" being recognized as a Date object, but an object with all of the other characteristics of a Date object not

In other words, for a type system built in to JS to be sufficiently valuable, it needs to actually be able to type all possible JS.

Thoughts on this? I'm sure there's lots of qualifiers to my stated goal here, but I hope this captures the gist.
